lib/stdlib/src/maps.erl

```diff
@@ -65,6 +67,58 @@ find(_,_) -> erlang:nif_error(undef).
 
 from_list(_) -> erlang:nif_error(undef).
 
+-spec intersect(Map1,Map2) -> Map3 when
+    Map1 :: #{Key => term()},
+    Map2 :: #{term() => Value2},
+    Map3 :: #{Key => Value2}.
+
+intersect(Map1, Map2) when is_map(Map1), is_map(Map2) ->
+    intersect_with(Map1,Map2,fun intersect_combiner/3);
+intersect(Map1, Map2) ->
+    erlang:error(error_type_two_maps(Map1, Map2),
+                 [Map1,Map2]).
+
+intersect_combiner(_K, _V1, V2) ->
+    V2.
+
+-spec intersect_with(Map1, Map2, Combiner) -> Map3 when
+    Map1 :: #{Key => Value1},
+    Map2 :: #{term() => Value2},
+    Combiner :: fun((Key, Value1, Value2) -> CombineResult),
+    Map3 :: #{Key => CombineResult}.
+
+intersect_with(Map1, Map2, Combiner) when is_map(Map1),
+                                          is_map(Map2),
+                                          is_function(Combiner, 3) ->
+    case map_size(Map1) < map_size(Map2) of
+        true ->
+            Iterator = maps:iterator(Map1),
+            intersect_with_1(maps:next(Iterator),
+                             Map1,
+                             Map2,
+                             Combiner);
+        false ->
+            Iterator = maps:iterator(Map2),
+            intersect_with_1(maps:next(Iterator),
+                             Map2,
+                             Map1,
+                             fun(K, V1, V2) -> Combiner(K, V2, V1) end)
+    end;
+intersect_with(Map1, Map2, Combiner) ->
+   error(error_type_merge_intersect_with(Map1, Map2, Combiner),
+         [Map1, Map2, Combiner]).
+
+intersect_with_1({K, _, Iterator}, Map1, Map2, Combiner) ->
+    case {Map1, Map2} of
+        {#{ K := V1 }, #{ K := V2 }} ->
```

Comment:
`maps:next/1` returns both key and value, so there's no need to look it up again:

```suggestion
intersect_with_1({K, V1, Iterator}, Map1, Map2, Combiner) ->
    case Map2 of
        #{ K := V2 } ->
```

Reply:
Thanks, fixed in 1a5f61a